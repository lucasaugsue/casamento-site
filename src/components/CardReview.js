import AttachMoneyIcon from '@mui/icons-material/AttachMoney';
import FavoriteIcon from '@mui/icons-material/Favorite';
import ShareIcon from '@mui/icons-material/Share';
import Avatar from '@mui/material/Avatar';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import Tooltip from '@mui/material/Tooltip';
import Typography from '@mui/material/Typography';
import { red } from '@mui/material/colors';
import moment from 'moment';
import 'moment/locale/pt-br';
import { forwardRef } from 'react';

const styleIconButton = {
  color: '#757575', 
  margin: '1vh 0.25vw', 
  cursor: 'pointer'
}

const CardReview = forwardRef(({params}, ref) => {
  const colors = [
    '#30567e', '#922f23', '#dbc78c', 
    '#ce674f', '#f49cb4', '#5d573a'
  ]

  return (
    <Card ref={ref} sx={{ maxWidth: 345, borderRadius: 5 }}>
      <CardHeader
        avatar={
          <Avatar sx={{ bgcolor: colors[params.color_id] || red[500] }} aria-label="recipe">
            {params.nome ? params.nome[0].toUpperCase() : ""}
          </Avatar>
        }
        title={params.nome || ""}
        subheader={params.created_at ? moment(params.created_at).format('MMMM Do YYYY, h:mm') : ""}
      />
      <CardMedia
        component="img"
        height="194"
        image={params.url || ""}
        alt={params.nome || ""}
        sx={{ cursor: 'pointer' }}
        onClick={() => {params.handleChange()}}
      />
      <CardContent>
        <Typography variant="body2" color="text.secondary">
          {
            (params.descricao && params.descricao.length > 150)
            ?`${params.descricao.slice(0, 146)}...`
            :params.descricao || ""
          }
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
        <Tooltip title="Favorito dos noivos">
          <div style={styleIconButton}>
            <FavoriteIcon sx={params.favorito ? { color: red[500] } : {} } />
          </div>
        </Tooltip>
        <Tooltip title="Comprar presente">
          <div style={styleIconButton} onClick={() => {params.handleChange()}}>
            <AttachMoneyIcon />
          </div>
        </Tooltip>
        <Tooltip title="Copiar link">
          <div style={styleIconButton} onClick={() => {params.compartilhar()}}>
            <ShareIcon />
          </div>
        </Tooltip>
      </CardActions>
    </Card>
  );
})

export default CardReview;