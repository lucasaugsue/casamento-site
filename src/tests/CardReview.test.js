import renderer from 'react-test-renderer';
import CardReview from '../components/CardReview';

const item = {
    "mais_informacoes": "https://www.amazon.com.br/dp/B001BSX1EM?ref_=cm_sw_r_apan_dp_20ZZWBH7X76K0ENAKVDQ&language=pt-BR",
    "descricao": "O núcleo de aço de calibre pesado não deforma e aquece uniformemente sem pontos quentes para que seus biscoitos e bolos saiam do forno perfeitamente e uniformemente dourados.\nO interior de cada peça tem camadas antiaderentes interligadas que são projetadas para proporcionar liberação duradoura e de alto desempenho. Libera até 2 vezes melhor do que as assadeiras clássicas Calphalon.",
    "created": "2023-12-13T19:25:40.385Z",
    "url": "https://m.media-amazon.com/images/I/81SJ7sZHKYL._AC_UF894,1000_QL80_FMwebp_.jpg",
    "updated": "2023-12-13T19:30:02.715Z",
    "preco": "494.50",
    "nome": "Calphalon Conjunto de assadeiras antiaderentes, conjunto de 6 peças",
    "id": "27ab93168ffb6730d4a0e62336d184e9"
}

test('Renderiza CardReview corretamente', async () => {
    const handleChange = jest.fn();
    const compartilhar = jest.fn();

    const tree = renderer.create(
        <CardReview 
            params={{
                ...item, 
                color_id: 1,
                handleChange: handleChange,
                compartilhar: compartilhar
            }} 
        />
    ).toJSON();

    // Verifica se o snapshot corresponde ao esperado
    expect(tree).toMatchSnapshot();
});
