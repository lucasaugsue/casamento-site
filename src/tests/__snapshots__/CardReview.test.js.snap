// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardReview/> Renderização deve renderizar CardReview corretamente 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1lq2m3b-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
    >
      <div
        aria-label="recipe"
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-nq71by-MuiAvatar-root"
      >
        C
      </div>
    </div>
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
      >
        Calphalon Conjunto de assadeiras antiaderentes, conjunto de 6 peças
      </span>
      <span
        className="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
      />
    </div>
  </div>
  <img
    alt="Calphalon Conjunto de assadeiras antiaderentes, conjunto de 6 peças"
    className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-17ibesn-MuiCardMedia-root"
    height="194"
    onClick={[Function]}
    src="https://m.media-amazon.com/images/I/81SJ7sZHKYL._AC_UF894,1000_QL80_FMwebp_.jpg"
  />
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      O núcleo de aço de calibre pesado não deforma e aquece uniformemente sem pontos quentes para que seus biscoitos e bolos saiam do forno perfeitamen...
    </p>
  </div>
  <div
    className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
  >
    <div
      aria-label="Favorito dos noivos"
      aria-labelledby={null}
      className=""
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        {
          "color": "#757575",
          "cursor": "pointer",
          "margin": "1vh 0.25vw",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="FavoriteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        />
      </svg>
    </div>
    <div
      aria-label="Comprar presente"
      aria-labelledby={null}
      className=""
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        {
          "color": "#757575",
          "cursor": "pointer",
          "margin": "1vh 0.25vw",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AttachMoneyIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.8 10.9c-2.27-.59-3-1.2-3-2.15 0-1.09 1.01-1.85 2.7-1.85 1.78 0 2.44.85 2.5 2.1h2.21c-.07-1.72-1.12-3.3-3.21-3.81V3h-3v2.16c-1.94.42-3.5 1.68-3.5 3.61 0 2.31 1.91 3.46 4.7 4.13 2.5.6 3 1.48 3 2.41 0 .69-.49 1.79-2.7 1.79-2.06 0-2.87-.92-2.98-2.1h-2.2c.12 2.19 1.76 3.42 3.68 3.83V21h3v-2.15c1.95-.37 3.5-1.5 3.5-3.55 0-2.84-2.43-3.81-4.7-4.4z"
        />
      </svg>
    </div>
    <div
      aria-label="Copiar link"
      aria-labelledby={null}
      className=""
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        {
          "color": "#757575",
          "cursor": "pointer",
          "margin": "1vh 0.25vw",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ShareIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
        />
      </svg>
    </div>
  </div>
</div>
`;
